File: Production-SPA/src/app/models/annual-production-plan.ts
import { Worker } from '../models/worker';
import { PlanItem } from './plan-item';

export class AnnualPlan {
  id: number;
  dateOfIssue: Date;
  expirationDate: Date;
  description: string;
  note: string;
  workerId: number;
  worker: Worker;
  planItems: PlanItem[];

  constructor(p?: any) {
    if (p == null) {
      return;
    }
    this.id = p.id;
    this.dateOfIssue = p.dateOfIssue;
    this.expirationDate = p.expirationDate;
    this.description = p.description;
    this.note = p.note;
    this.workerId = p.workerId;
    this.worker = new Worker(p.UnitOfMeasure);
    this.planItems = new Array(p.planItems);
  }
}

File: Production Back/Production.Domen/AnnualProductionPlan.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Production.Domen
{
    public class AnnualProductionPlan
    {
        public int Id { get; set; }
        public DateTime DateOfIssue { get; set; }
        public DateTime ExpirationDate { get; set; }
        public string Description { get; set; }
        public string Note { get; set; }
        public int WorkerId { get; set; }
        public Worker Worker { get; set; }
        public ICollection<PlanItem> PlanItems { get; set; }

    }
}
